src/Symfony/Component/EventDispatcher/DependencyInjection/RegisterListenersPass.php

```diff
@@ -115,8 +115,12 @@ public function process(ContainerBuilder $container)
                     ], function ($matches) { return strtoupper($matches[0]); }, $event['event']);
                     $event['method'] = preg_replace('/[^a-z0-9]/i', '', $event['method']);
 
-                    if (null !== ($class = $container->getDefinition($id)->getClass()) && ($r = $container->getReflectionClass($class, false)) && !$r->hasMethod($event['method']) && $r->hasMethod('__invoke')) {
-                        $event['method'] = '__invoke';
+                    if (null !== ($class = $container->getDefinition($id)->getClass()) && ($r = $container->getReflectionClass($class, false)) && !$r->hasMethod($event['method'])) {
+                        if ($r->hasMethod('__invoke')) {
+                            $event['method'] = '__invoke';
+                        } elseif (!$r->hasMethod($event['method'])) {
+                            throw new InvalidArgumentException(sprintf('Service "%s" must define the "method" attribute on "%s" tags.', $id, $this->listenerTag));
```

Comment:
You can add that none of `$event['method']` and `__invoke` are defined. That would help to discover the default naming conventions and eventually track a typo in the method name.